Admin/AdminHelper.php

```diff
@@ -201,4 +228,113 @@ public function camelize($property)
+        $associationAdmin = $admin->getFormFieldDescription($element)->getAssociationAdmin();
+        if (count($elements) == 0) {
+            return $associationAdmin->getClass();
+        } else {
+            return $this->getEntityClassName($associationAdmin, $elements);
+        }
+    }
+
+    /**
+     * get access path to element which works with PropertyAccessor.
+     *
+     * @param string $elementId expects string in format used in form id field. (uniqueIdentifier_model_sub_model or uniqueIdentifier_model_1_sub_model etc.)
+     * @param mixed  $entity
+     *
+     * @return string
+     *
+     * @throws \Exception
+     */
+    public function getElementAccessPath($elementId, $entity)
+    {
+        $propertyAccessor = new PropertyAccessor();
+
+        $id_without_unique_identifier = implode('_', explode('_', substr($elementId, strpos($elementId, '_') + 1)));
+
+        //array access of id converted to format which PropertyAccessor understands
+        $initial_path = preg_replace('#(_(\d+)_)#', '[$2]', $id_without_unique_identifier);
+
+        $parts = preg_split('#\[\d+\]#', $initial_path);
+
+        $part_return_value = $return_value = '';
+        $current_entity = $entity;
+
+        foreach ($parts as $key => $value) {
+            $sub_parts = explode('_', $value);
+            $id = '';
+            $dot = '';
+
+            foreach ($sub_parts as $sub_value) {
+                $id .= ($id) ? '_'.$sub_value : $sub_value;
+
+                if ($this->pathExists($propertyAccessor, $current_entity, $part_return_value.$dot.$id)) {
+                    $part_return_value .= $dot.$id;
+                    $dot = '.';
+                    $id = '';
+                } else {
+                    $dot = '';
+                }
+            }
+
+            if ($dot !== '.') {
+                throw new \Exception(sprintf('Could not get element id from %s Failing part: %s', $elementId, $sub_value));
+            }
+
+            //check if array access was in this location originally
+            preg_match("#$value\[(\d+)#", $initial_path, $matches);
+
+            if (isset($matches[1])) {
+                $part_return_value .= '['.$matches[1].']';
+            }
+
+            $return_value .= $return_value ? '.'.$part_return_value : $part_return_value;
+            $part_return_value = '';
+
+            if (isset($parts[$key + 1])) {
+                $current_entity = $propertyAccessor->getValue($entity, $return_value);
+            }
+        }
+
+        return $return_value;
+    }
+
+    /**
+     * check if given path exists in $entity.
+     *
+     * @param PropertyAccessor $propertyAccessor
+     * @param mixed            $entity
+     * @param string           $path
+     *
+     * @return bool
+     *
+     * @throws \RuntimeException
+     */
+    private function pathExists(PropertyAccessor $propertyAccessor, $entity, $path)
+    {
+        //sf2 <= 2.3 did not have isReadable method for PropertyAccessor
```

Comment:
not sure if need this @core23 @greg0ire , because we don't support SF 2.3 in the current version anymore